mkosi/archive.py

```diff
@@ -102,3 +104,21 @@ def make_cpio(src: Path, dst: Path, files: Optional[Iterable[Path]] = None) -> N
         # Make sure tar uses user/group information from the root directory instead of the host.
         options=finalize_passwd_mounts(dst),
     )
+
+
+def normalize_mtime(root: Path, source_date_epoch: Optional[str]) -> None:
+    if source_date_epoch is None:
+        return
+
+    try:
+        mtime_epoch = int(source_date_epoch)
```

Comment:
This is probably good enough, but considering the nanosecond resolution used below, wouldn't a float be valid as well? Does this make sense? Are there people using subsecond resolution for this?

Reply:
After looking at this again, I'm pretty sure SOURCE_DATE_EPOCH is meant to be an integer:
> The value MUST be an ASCII representation of an integer with no fractional component, identical to the output format of date +%s.


https://reproducible-builds.org/specs/source-date-epoch/